src/cunumeric/index/zip.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/zip.h"
+#include "cunumeric/index/zip_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <int DIM, int N>
+struct ZipImplBody<VariantKind::CPU, DIM, N> {
+  using VAL = int64_t;
+
+  template <size_t... Is>
+  void operator()(const AccessorWO<Point<N>, DIM>& out,
+                  const std::vector<AccessorRO<VAL, DIM>>& index_arrays,
+                  const Rect<DIM>& rect,
+                  const Pitches<DIM - 1>& pitches,
+                  bool dense,
+                  const int64_t key_dim,
+                  const int64_t start_index,
+                  std::index_sequence<Is...>) const
+  {
+    if (index_arrays.size() == N) {
+      const size_t volume = rect.volume();
+      if (dense) {
+        std::vector<const VAL*> indx_ptrs;
+        for (auto a : index_arrays) indx_ptrs.push_back(a.ptr(rect));
```

Comment:
Minor, but I would do `for (auto& a` here. I believe the current `for (auto a` will be interpreted as `for (AccessorRO a`, which will create an intermediate copy of each element.

Reply:
Since you're doing the `index_sequence` trick, I think you could also use this here as well!
```
std::vector<const VAL*> indx_ptrs = {index_arrays[Is].ptr(rect)...};
```